osc.js

```diff
@@ -307,8 +446,353 @@ class OSCInstance extends InstanceBase {
+						return comparisonResult;
+					} else {
+						this.log('debug', `Feedback ${feedback.id} returned false! Path does not exist yet in dictionary.`);
+						return false;
+					}
+				}
+			},
+			osc_feedback_multi: {
+				type: 'boolean',
+				name: 'Listen for OSC messages (Multiple Arguments)',
+				description: 'Listen for OSC messages. Requires "Listen for Feedback" option to be enabled in OSC config.',
+				options: [
+					{
+						type: 'textinput',
+						label: 'OSC Path',
+						id: 'path',
+						default: '/osc/path',
+						useVariables: true,
+					},
+					{
+						type: 'textinput',
+						label: 'Arguments',
+						id: 'arguments',
+						default: '1 "test" 2.5',
+						useVariables: true,
+					},
+					{
+						id: 'comparison',
+						type: 'dropdown',
+						label: 'Comparison',
+						choices: [
+							{ id: 'equal', label: '=' },
+							{ id: 'notequal', label: '!=' },
+						],
+						default: 'equal'
+					}
+				],
+				callback: async (feedback, context) => {
+					const path = await context.parseVariablesInString(feedback.options.path || '');
+					let argsStr = await context.parseVariablesInString(feedback.options.arguments || '');
+					const comparison = feedback.options.comparison;
+			
+					this.log('debug', `Evaluating feedback ${feedback.id}.`);
+			
+					const { args, error } = parseArguments(argsStr);
+					if (error) {
+						this.log('warn', error);
+						return false;
+					}
+			
+					if (this.onDataReceived.hasOwnProperty(path)) {
+						const rx_args = this.onDataReceived[path];
+						let comparisonResult = (comparison === 'equal');
+						for (let i = 0; i < args.length; i++) {
+							comparisonResult = evaluateComparison(rx_args[i], args[i], comparison);
+							if ((comparison === 'equal' && !comparisonResult) || (comparison === 'notequal' && comparisonResult)) {
+								break;
+							}
+						}
+			
+						this.log('debug', `Feedback ${feedback.id} comparison result: ${comparisonResult}`);
+						return comparisonResult;
+					} else {
+						this.log('debug', `Feedback ${feedback.id} returned false! Path does not exist yet in dictionary.`);
+						return false;
+					}
+				}
+			},			
+			osc_feedback_noargs: {
+				type: 'boolean',
+				name: 'Listen for OSC messages (No Arguments)',
+				description: 'Listen for OSC messages. Requires "Listen for Feedback" option to be enabled in OSC config.',
+				options: [
+					{
+						type: 'textinput',
+						label: 'OSC Path',
+						id: 'path',
+						default: '/osc/path',
+						useVariables: true,
+					}
+				],
+				callback: async (feedback, context) => {
+					const path = await context.parseVariablesInString(feedback.options.path || '');
+					this.log('debug', `Evaluating feedback ${feedback.id}.`);
+	
+					if (this.onDataReceived.hasOwnProperty(path)) {
+						this.log('debug', `Feedback ${feedback.id} returned true!`);
+						delete this.onDataReceived[path]; // Remove the path from the dictionary to create a debounce
```

Comment:
I'm not sure about this, companion can re-check a feedback at any time, which could result in this causing values to disappear at random

Reply:
This is a specific implentation of "osc_feedback_noargs" feedback, as there are no arguments to otherwise compare against to validate the feedback - the command being received is the trigger. The way I implemented this was when the command gets receieved, it will cause a "debounce".

For example:

If I send /osc/test1 with the feedback configured accordingly, sending it will trigger the feedback.

If I then send /osc/test2 or any other OSC command, it will release the feedback for /osc/test1 and trigger the next one. Using the onDataReceived array is what tracks this.

Reply:
I've added an additional check (an OSC message has indeed arrived) to the if statement to help mitigate this.